### PR TITLE
Using New Docker Client Python Package

### PR DESCRIPTION
**Pull request by keitwb**

### CODE REVIEW

File: dockerplugin.py [hunk 1]
```diff
@@ -198,11 +198,14 @@ def read_cpu_stats(container, dimensions, stats, t):
 
 def read_network_stats(container, dimensions, stats, t):
     """Process network utilization stats for a container."""
-    net_stats = stats['network']
+    net_stats = stats['networks']
```
Comment (mpetazzoni) [inline dockerplugin.py:201]: I'm assuming that's what's returned by the client library regardless of the version of the Docker daemon you talk to, as long as it uses API v1.21 or above?

File: dockerplugin.py [hunk 2]
```diff
@@ -198,11 +198,14 @@ def read_cpu_stats(container, dimensions, stats, t):
...
     log.info('Reading network stats: {0}'.format(net_stats))
 
-    items = sorted(net_stats.items())
-    emit(container, dimensions, 'network.usage', [x[1] for x in items], t=t)
+    for interface, if_stats in net_stats.items():
+        items = sorted(if_stats.items())
+        interface_dims = dimensions.copy()
+        interface_dims['interface'] = interface
```
Comment (mpetazzoni) [inline dockerplugin.py:207]: That's a change in behavior of the metrics reporting, so we'll need a major version change for it. We'll also have to make sure that the Docker dashboards work correctly with those new timeseries.

**Review by mpetazzoni — APPROVED** [review]
LGTM, but see comments for caveats

File: dockerplugin.py (hunk 1 above)
Comment (keitwb) [reply dockerplugin.py:201]: Yes, the client uses versioned endpoints to the docker daemon so the output is always in this format.

File: dockerplugin.py (hunk 2 above)
Comment (keitwb) [reply dockerplugin.py:207]: Dashboards looked good when I checked -- only change is the addition of an `interface` dimension, which is ignored by the built in content.  They didn't change the metric names themselves in 1.21.

**Pull request merged**